### PR TITLE
Polar kinematics

### LINKED ISSUE #115: Implement polar versions of kinematic vectors
Continuing discussion from #106

- start vector utils module, including for now `cart2polar` and `polar2cart` functions.
- Add the versions of the data variables in polar coordinates: `position_polar`, `velocity_polar`, `accelaration_polar`. Instead of a `space` axis, these polar variables will have a `space_polar` axis, with two coordinates `rho` and `phi.`

### PR DESCRIPTION
**Pull request by lochhh**
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
This PR closes #115.

**What does this PR do?**
- This PR adds the `utils.vector` module with functions to transform between Cartesian and polar coordinates.
- Using the aforementioned module, polar versions of the four kinematic properties (i.e. `position_pol`, `displacement_pol`, `velocity_pol`, and `acceleration_pol`) have been added to the `move_accessor` . 
- For convenience, the `position` property has also been added to the `move_accessor`. 

## References
#115

## How has this PR been tested?
Unit and integration tests have been added and run successfully.

## Does this PR require an update to the documentation?
Module has been added to the docs.

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Review by niksirbi — APPROVED** [review]
Thanks @lochhh!

This is very well implemented, I only had a few questions/comments.

The new features are not currently documented in any example/guide, so I opened an [issue](https://github.com/neuroinformatics-unit/movement/issues/160) as a reminder for us to do that before the next release.

Other than that, there is the question of agreeing on a consistent interface for accessing and storing kinematic variables, which we discussed during our [last meeting](https://neuroinformatics.zulipchat.com/#narrow/stream/406001-Movement/topic/Meeting.20Minutes/near/431503639). But I propose to merge this PR before we make a final decision on that topic. I will make sure to open a separate issue to discuss that in detail.

File: movement/move_accessor.py [hunk 1]
```diff
@@ -128,6 +129,49 @@ def acceleration(self) -> xr.DataArray:
...
+            The computed property in polar coordinates.
+        """
+        if property not in self._obj:
+            self._obj[property] = vector.cart2pol(
+                getattr(self, property.replace("_pol", ""))
+            )
+        return self._obj[property]
+
+    @property
+    def displacement_pol(self) -> xr.DataArray:
```
[the remark this answers is not in the document]
Comment (lochhh) [reply movement/move_accessor.py:153]: Good catch! This is an oversight and should have been `space_pol`. If we decide to go with "polar", I'll have to rename everything with "pol", including fixtures (e.g. `cart_pol_dataset_missing_pol_dim` to `cart_polar_dataset_missing_polar_dim`), functions (e.g. `cart2pol` to `cart2polar`).

File: movement/move_accessor.py [hunk 2]
```diff
@@ -128,6 +129,49 @@ def acceleration(self) -> xr.DataArray:
...
+        Returns
+        -------
+        xarray.DataArray
+            The computed property in polar coordinates.
+        """
+        if property not in self._obj:
+            self._obj[property] = vector.cart2pol(
+                getattr(self, property.replace("_pol", ""))
+            )
+        return self._obj[property]
```
[the remark this answers is not in the document]
Comment (lochhh) [reply movement/move_accessor.py:150]: > What happend if `ds.move.velocity_polar` is called before `ds.move.velocity`? I assume the cartesian `velocity` property will be imlplicitly computed and stored?

Yes. The `getattr` in L148 tries to retrieve the cartesian property via the property accessor. So both the cart and pol properties will be stored in the Dataset.

 

> Also, im theory, we could also have `position_polar`, right? Should we provide that option?

Sure.

File: movement/move_accessor.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (niksirbi) [reply movement/move_accessor.py:153]: Hmm, I'm fine with either "pol" or "polar" as long as it's consistent, but have a slight personal preference for "polar". I'll leave it up to you to decide, it's not super critical.

**Pull request merged**